File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    
    d = {}
    for [name, kind] in clothes:
        if kind not in d:
            d[kind] = 1
        else:
            d[kind] += 1
   
    for key in d:
        answer *= d[key]+1
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, HashSet<String>> map = new HashMap<>();
        for(int i=0; i<clothes.length; i++) {
            for(int j=0; j<2; j++) {
                if (map.containsKey(clothes[i][1])) {
                    map.get(clothes[i][1]).add(clothes[i][0]);
                } else {
                    map.put(clothes[i][1], new HashSet<String>());
                    map.get(clothes[i][1]).add(clothes[i][0]);
                }
                
            }
        }
        for (String key: map.keySet()) {
            answer *= (map.get(key).size() + 1);
        }
        return answer - 1;
    }
}